install-and-configure/install/multi-cluster/federated-etl/thanos-migration-guide.md

```diff
@@ -0,0 +1,146 @@
+
+If this is your first time setting up an object store, refer to these docs:
+
+* [AWS](/install-and-configure/install/multi-cluster/long-term-storage-configuration/long-term-storage-aws.md)
+* [Azure](/install-and-configure/install/multi-cluster/long-term-storage-configuration/long-term-storage-azure.md)
+* [GCP](/install-and-configure/install/multi-cluster/long-term-storage-configuration/long-term-storage-gcp.md)
+
+### Step 2: Enable ETL backups on the *primary cluster only*
+
+Enabling [ETL backups](/install-and-configure/install/etl-backup/etl-backup.md) ensures Kubecost persists historical data in durable storage (outside of Thanos) and stores the data in a format consumable by the ETL Utils container. The ETL Utils container transforms that data and writes it to a separate location in the object store for consumption by Aggregator.
+
+```yaml
+kubecostModel:
+  etlBucketConfigSecret: <YOUR_SECRET_NAME>
+```
+
+### Step 3: Validate that an `/etl` directory is present in the object store
+
+There should be ETL data present in the following directories. CloudCosts will only have ETL data if at least one cloud integration is enabled.
+
+* `/etl/bingen/allocations`
+* `/etl/bingen/assets/`
+* `/cloudcosts/`
+
+### Step 4: Create a new federated-store secret on the primary cluster
+
+This will point to the existing Thanos object store or the new object store created in Step 1. The secret should be identical to your `object-store.yaml`, with the exception that this new secret *must* be named `federated-store.yaml`.
+
+{% hint style="warning" %}
+The name of the .yaml file used to create the secret *must* be named *federated-store.yaml* or Aggregator will not start.
+{% endhint %}
+
+```sh
+kubectl create secret generic federated-store --from-file=federated-store.yaml -n kubecost
+```
+
+### Step 5: Set Aggregator and ETL Utils in *values.yaml* on the primary cluster
+
+{% hint style="warning" %}
+*Do not* disable Thanos during this step. This will not negatively impact the source data. Thanos will be disabled in a later step.
+{% endhint %}
+
+* [Enable Aggregator](https://docs.kubecost.com/install-and-configure/install/multi-cluster/federated-etl/aggregator) for the primary cluster. Your ETL Utils configuration should look like:
+
+```yaml
+etlUtils:
+  enabled: true
+  thanosSourceBucketSecret: kubecost-thanos
+  fullImageName: gcr.io/kubecost1/cost-model-etl-utils:0.2
+  resources: {}
+kubecostModel:
+  federatedStorageConfigSecret: federated-store
+```
+
+### Step 6: Apply the changes and wait for data to populate in the UI
+
+Ensure all pods and containers are running:
+
+```txt
+kubecost-cost-analyzer-685fd8f677-k652h        4/4     Running   2 (12m ago)   3h2m
+kubecost-forecasting-6cf4dd7b98-7z8f5          1/1     Running   0             66m
+kubecost-etl-utils-6cdd489596-5dl75           1/1     Running   0          6d20h
+```
+
```

Comment:
There's another thread open for this, but I want to make doubly-sure we address it: this is not the expected state. It's more like this:

```
→ k get pod -n kubecost-cloudcost
NAME                                                    READY   STATUS    RESTARTS     AGE
kubecost-cloudcost-aggregator-0                         1/1     Running   0            16h
kubecost-cloudcost-etl-utils-79b89f66-v5jq2             1/1     Running   0            3d22h
kubecost-cloudcost-cost-analyzer-7f9b79d74-rmrrt        2/2     Running   0            3d16h
```

Reply:
@michaelmdresser Something I'll clarify in this PR is that if you're migrating off Thanos, you need to perform all these steps before even upgrading up to 2.0. This is necessary since Thanos is not included in the 2.0 Helm Chart. Therefore the 4 containers in `kubecost-cost-analyzer` in v1.108 may be the expected state? I don't expect forecasting though.

I'm going to focus on refining & publishing this doc today. I'll update this code block!

Reply:
Great clarification, that makes excellent sense